base/linalg/symmetric.jl

```diff
@@ -117,8 +117,16 @@ A_mul_B!{T<:BlasComplex,S<:StridedMatrix}(C::StridedMatrix{T}, A::StridedMatrix{
 *(A::HermOrSym, B::HermOrSym) = full(A)*full(B)
 *(A::StridedMatrix, B::HermOrSym) = A*full(B)
 
-factorize(A::HermOrSym) = bkfact(A.data, symbol(A.uplo), issym(A))
+bkfact(A::HermOrSym) = bkfact(A.data, symbol(A.uplo), issym(A))
+factorize(A::HermOrSym) = bkfact(A)
```

Comment:
Okay, what I am really after is that my interpretation of the [documentation](https://github.com/JuliaLang/julia/blob/7affd0ced06c3596680a7ad69081aa7b98ba6ef9/base/linalg/dense.jl#L662-L683) is that this should return a `Cholesky` factorization since `H` is positive definite:

```julia
julia> A = rand(3, 3); A = A'A; H = Hermitian(A);

julia> isposdef(H)
true

julia> typeof(factorize(H))
Base.LinAlg.BunchKaufman{Float64,Array{Float64,2}}
```